homeassistant/components/mqtt/client.py

```diff
@@ -907,11 +910,13 @@ async def _discovery_cooldown(self) -> None:
         wait_until = max(
             last_discovery + DISCOVERY_COOLDOWN, last_subscribe + DISCOVERY_COOLDOWN
         )
-        while now < wait_until:
+        while now < wait_until or self._pending_subscriptions:
             await asyncio.sleep(wait_until - now)
             now = time.time()
             last_discovery = self._mqtt_data.last_discovery
-            last_subscribe = self._last_subscribe
+            last_subscribe = (
```

Comment:
Mirror this in the pre-while condition.

Reply:
I am not going to change the current code that is not related.